qadence/ml_tools/saveload.py

```diff
@@ -58,8 +58,18 @@ def write_checkpoint(
     from qadence.ml_tools.models import TransformedModule
     from qadence.models import QNN, QuantumModel
 
-    model_checkpoint_name: str = f"model_{type(model).__name__}_ckpt_" + f"{iteration:03n}" + ".pt"
-    opt_checkpoint_name: str = f"opt_{type(optimizer).__name__}_ckpt_" + f"{iteration:03n}" + ".pt"
+    device = None
+    try:
+        # We extract the device from the pyqtorch native circuit
+        device = model.device
```

Comment:
```suggestion
        device = str(model.device).split(":")[0]
```
or sth like this?
To make sure we only get `cuda` and not `cuda:0` as a name?

Reply:
```
awennersteen:~$ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.device("cuda:0")
device(type='cuda', index=0)
>>> a = torch.device("cuda:0")
>>> str(a)
'cuda:0'
>>> str(a).split(":")[0]
'cuda'
>>> 
```

Reply:
great, thanks